cmd/soroban-cli/src/commands/contract/deploy/wasm.rs

```diff
@@ -100,19 +108,91 @@ pub enum Error {
     Network(#[from] network::Error),
     #[error(transparent)]
     Wasm(#[from] wasm::Error),
+    #[error("cannot access config dir for alias file")]
+    CannotAccessConfigDir,
+    #[error(
+        "alias must be 1-30 chars long, and have only letters, numbers, underscores and dashes"
+    )]
+    InvalidAliasFormat { alias: String },
+    #[error(transparent)]
+    JSONSerialization(#[from] serde_json::Error),
+    #[error(transparent)]
+    IO(#[from] std::io::Error),
 }
 
 impl Cmd {
     pub async fn run(&self) -> Result<(), Error> {
+        self.validate_alias()?;
+
         let res = self.run_against_rpc_server(None, None).await?.to_envelope();
         match res {
             TxnEnvelopeResult::TxnEnvelope(tx) => println!("{}", tx.to_xdr_base64(Limits::none())?),
             TxnEnvelopeResult::Res(contract) => {
+                self.save_contract_id(&contract)?;
                 println!("{contract}");
             }
         }
         Ok(())
     }
+
+    fn validate_alias(&self) -> Result<(), Error> {
+        let alias = self.alias();
+
+        if alias.is_empty() {
+            return Ok(());
+        }
+
+        let regex = Regex::new(r"^[a-zA-Z0-9_-]{1,30}$").unwrap();
+
+        if regex.is_match(&alias) {
+            Ok(())
+        } else {
+            Err(Error::InvalidAliasFormat { alias })
+        }
+    }
+
+    fn alias(&self) -> String {
+        self.alias.as_ref().map(Clone::clone).unwrap_or_default()
+    }
+
+    fn alias_path(&self) -> Result<PathBuf, Error> {
+        let config_dir = self.config.config_dir()?;
+        let network = self.config.network.network.clone().expect("must be set");
+        let alias = self.alias();
+        let file_name = format!("{alias}.json");
+
+        Ok(config_dir
+            .join("contract-ids")
+            .join(network)
+            .join(file_name))
+    }
+
+    fn save_contract_id(&self, contract: &String) -> Result<(), Error> {
+        let file_path = self.alias_path()?;
+
+        let Some(dir) = file_path.parent() else {
+            return Err(Error::CannotAccessConfigDir);
+        };
+
+        match create_dir_all(dir) {
+            Ok(()) => {}
+            _ => return Err(Error::CannotAccessConfigDir),
+        };
+
+        let mut to_file = OpenOptions::new()
+            .create(true)
```

Comment:
Clippy is telling me this line has undefined behaviour because it isn't accompanied by `.truncate(bool)` to indicate if the file should be truncated or writing should be appended. I think this should be truncated because the goal is to write over:
```suggestion
            .create(true)
            .truncate(true)
```

Reply:
interesting… my clippy is running clean!

```console
$ cargo clippy -- -Dwarnings -Dclippy::all -Dclippy::pedantic
warning: /Users/fnando/Projects/stellar/stellar-cli/cmd/stellar-cli/Cargo.toml: file `/Users/fnando/Projects/stellar/stellar-cli/cmd/stellar-cli/src/bin/main.rs` found to be present in multiple build targets:
  * `bin` target `stellar`
  * `bin` target `soroban`
warning: /Users/fnando/Projects/stellar/stellar-cli/cmd/soroban-cli/Cargo.toml: file `/Users/fnando/Projects/stellar/stellar-cli/cmd/soroban-cli/src/bin/main.rs` found to be present in multiple build targets:
  * `bin` target `stellar`
  * `bin` target `soroban`
   Compiling soroban-cli v21.0.0-rc.1 (/Users/fnando/Projects/stellar/stellar-cli/cmd/soroban-cli)
    Checking soroban-spec-typescript v21.0.0-rc.1 (/Users/fnando/Projects/stellar/stellar-cli/cmd/crates/soroban-spec-typescript)
    Checking soroban-spec-tools v21.0.0-rc.1 (/Users/fnando/Projects/stellar/stellar-cli/cmd/crates/soroban-spec-tools)
    Checking soroban-spec-json v21.0.0-rc.1 (/Users/fnando/Projects/stellar/stellar-cli/cmd/crates/soroban-spec-json)
    Checking soroban-test v21.0.0-rc.1 (/Users/fnando/Projects/stellar/stellar-cli/cmd/crates/soroban-test)
    Finished dev [unoptimized + debuginfo] target(s) in 2.78s
```